include/crpropa/magneticField/PolarizedSingleModeMagneticField.h

```diff
@@ -0,0 +1,67 @@
+#ifndef CRPROPA_POLARIZEDSINGLEMODEMAGNETICFIELD_H
+#define CRPROPA_POLARIZEDSINGLEMODEMAGNETICFIELD_H
+
+#include "crpropa/magneticField/MagneticField.h"
+
+#include <cstdlib>
+
+#include "crpropa/Vector3.h"
+#include "crpropa/Referenced.h"
+#include "crpropa/Units.h"
+
+#include <stdexcept>
+
+namespace crpropa {
+
+/**
+
+@class PolarizedSingleModeMagneticField
+@brief General poralized single mode magnetic field (including linear, circular and elliptic polarizations)
+
+
+*/
+
+class PolarizedSingleModeMagneticField: public MagneticField {
+private:
+	double B_0; // Magnetic field strength in the direction of e_1 at r_0
+	double wavelength; // Wavelength of the single mode (corresponds to its coherence length)
+	Vector3d r_0; // Reference position
+	Vector3d e_1; // First vector spanning the polarization plane
+	Vector3d e_2; // Second vector spanning the polarization plane 
+
+public:
+/** Constructor
+	@param B_0 // Magnetic field strength in the direction of e_1 at r_0
+	@param wavelength; // Wavelength of the single mode (corresponds to its coherence length)
+	@param r_0; // Reference position
+	@param e_1; // First vector spanning the polarization plane 
+	@param e_2; // Second vector spanning the polarization plane
+*/
+	PolarizedSingleModeMagneticField(double B_0, double wavelength, Vector3d r_0, Vector3d e_1, Vector3d e_2);
+
+	Vector3d getField(const Vector3d &pos, const double &sigma) const;
+
+	Vector3d getMaxHelFracField(const Vector3d &pos, const double &fH, const std::string &Bflag) const;
+	Vector3d getOrthogonalMaxHelFracField(const Vector3d &pos, const double &fH, const std::string &Bflag) const;
```

Comment:
These names are kind of long (which is fine), but it is hard to guess immediately what 'Frac' means, for example. Isn't there a way to explicitly write these things while keeping the function name relatively short?

Reply:
I could extend "Frac" to Fraction... My idea was that since fH is explained in the paper, it would be clear that HelFrac is related to that.

Reply:
In the new, completely renewed version, this is not an issue anymore.